ssl/statem/extensions_srvr.c

```diff
@@ -1753,13 +1753,22 @@ EXT_RETURN tls_construct_stoc_status_request(SSL_CONNECTION *s, WPACKET *pkt,
                                              unsigned int context, X509 *x,
                                              size_t chainidx)
 {
+    OCSP_RESPONSE *resp;
+
     /* We don't currently support this extension inside a CertificateRequest */
     if (context == SSL_EXT_TLS1_3_CERTIFICATE_REQUEST)
         return EXT_RETURN_NOT_SENT;
 
     if (!s->ext.status_expected)
         return EXT_RETURN_NOT_SENT;
 
+    /* Try to retrieve OCSP response for the actual certificate */
+    resp = get_ocsp_response(s, (int)chainidx);
+
+    /* If no OCSP response was found the extension is not sent */
```

Comment:
If I look at this function in 3.5 it would do the WPACKET_PUT_BYTES_u16 for a chainidx of zero and will also call the WPACKET_close. This is even if there was no OCSP Response. In this case none of that would be writing. Is this breaking something or was this a bug in 3.5.

https://github.com/openssl/openssl/blob/openssl-3.5/ssl/statem/extensions_srvr.c

Reply:
No, this is the very cause of the regression. You are not allowed to send an empty OCSP response in either a CertificateStatus message (TLS 1.2) or a CertificateEntry's `status_request` extension, full stop. If you do not have an OCSP response, you are *required* to omit the message/extension. See RFC 6066.

The old code looked like it did as you suggest, but it did not. When there was no OCSP response, `status_expected` would be false. Had the 3.5 code you're asking about ever run, there would be a bug, but it never ran.
https://github.com/openssl/openssl/blob/openssl-3.5/ssl/statem/statem_srvr.c#L2196

Then 3.6 introduced multi-stapling, which broke that assumption, which then triggered the buggy 3.5 code you're asking about. See also https://github.com/openssl/openssl/issues/28989